### PR TITLE
More precise *Field arguments and instance attributes

### PR DESCRIPTION
**Pull request by intgr**
Field constructors are quite inconsistent in Django REST Framework: some take position arguments, some don't. Some pass positional arguments to their parent class, others pass only keyword arguments. It took a while but I sorted through all the constructors and nailed down the precise behavior.

The stubs did not previously denote keyword-only arguments, and the order of positional arguments was slightly wrong in many places.

Also, added many instance attributes of *Field classes that were missing.

### CODE REVIEW

**Comment by intgr** [comment]
Pushed a small update, I found that `ChoiceField.choices` is actually a property that's always an `OrderedDict`.

File: rest_framework-stubs/fields.pyi [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import datetime
 import decimal
 import uuid
+from collections import OrderedDict
```
Comment (sobolevn) [inline rest_framework-stubs/fields.pyi]: Should we explicitly specify that it is `OrderedDict` and not just `dict`?
I guess that they have the same API.

I would be against this change.

Comment (intgr) [reply rest_framework-stubs/fields.pyi]: Using `Dict[]` is fine. Pushed an update for this.

I suppose `OrderedDict` is on its way to deprecation anyway, now that CPython's regular dict is now order-preserving.

**Comment by sobolevn** [comment]
Thanks!

**Comment by intgr** [comment]
Thanks too, working on TypedDjango projects has been pleasant. :)

**Pull request merged**